File: csharp/resistor-color/ResistorColor.cs
using System;
using System.Linq;

public static class ResistorColor
{
    private static readonly string[] _colors = new[] { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };

    public static int ColorCode(string color) => _colors.ToList().FindIndex(c => c == color);

    public static string[] Colors() => _colors;
}

File: c/resistor-color/resistor_color.c
#include "resistor_color.h"

static resistor_band_t all_colors[] = {BLACK, BROWN, RED, ORANGE, YELLOW, GREEN, BLUE, VIOLET, GREY, WHITE};

uint16_t color_code(resistor_band_t color)
{
    return color;
}

resistor_band_t *colors(void)
{
    return all_colors;
}
